ml/src/test/scala/frameless/ml/Generators.scala

```diff
@@ -0,0 +1,26 @@
+package frameless
+package ml
+
+import frameless.arbDouble
+import org.apache.spark.ml.linalg.{Matrices, Matrix, Vector, Vectors}
+import org.scalacheck.{Arbitrary, Gen}
+
+object Generators {
+
+  implicit val arbVector: Arbitrary[Vector] = Arbitrary {
+    val genDenseVector = Gen.listOf(arbDouble.arbitrary).map(doubles => Vectors.dense(doubles.toArray))
+    val genSparseVector = genDenseVector.map(_.toSparse)
+
+    Gen.oneOf(genDenseVector, genSparseVector)
+  }
+
+  implicit val arbMatrix: Arbitrary[Matrix] = Arbitrary {
+    Gen.sized { nbRows =>
```

Comment:
AFAIK `Gen.sized` uses maximum list length, so in this case, `nbRows` should be from `Gen.sized(Gen.choose(0, _))` see how `Gen.listOf` is implemented